tests/dialects/test_teradata.py

```diff
@@ -210,3 +210,72 @@ def test_cast(self):
                 "teradata": "TRYCAST('-2.5' AS DECIMAL(5, 2))",
             },
         )
+
+    def test_time(self):
```

Comment:
Let's also test teradata -> teradata to ensure the roundtrip is preserved for those tests where we only have eg. `read={"snowflake": ...}`. You can add `"teradata"` to the `read` dict as well.

Reply:
good idea, I added teradata to the `read` dict for all of these tests 👍